pkg/ibmvpc/parser.go

```diff
@@ -426,7 +426,7 @@ func getFipConfig(
 				// a node captured by a fip should not be captured by a pgw
 				for _, nodeWithFip := range srcNodes {
 					if vpcmodel.HasNode(pgw.Sources(), nodeWithFip) {
-						pgw.src = getCertainNodes(pgw.Sources(), func(n vpcmodel.Node) bool { return n.Cidr() != nodeWithFip.Cidr() })
+						pgw.src = getCertainNodes(pgw.Sources(), func(n vpcmodel.Node) bool { return n.CidrOrAddress() != nodeWithFip.CidrOrAddress() })
```

Comment:
Can the returned value here be indeed `cidr` or `address`?
(See general comment in this regard in `abstractVPC`)

Reply:
This will be changed in a follow up PR (#396 ), the comparison will be based on UID instead of address.

Reply:
Also in other places we are calling the function CidrOrAddress() but expect only Cidr?

Reply:
Yes, for example in `isEntirePublicInternetRange` within `externalNetwork.go`. The input is a list of nodes, but all are expected to be external nodes. 
One option is to convert `Node` to `ExternalNetwork`, and use its `CidrStr` field.
If the type `groupedExternalNodes` was defined as `[]ExternalNetwork` instead of `[]Node`, then conversion would not be required in this case.